iceoryx_utils/include/iceoryx_utils/internal/cxx/expected.inl

```diff
@@ -346,6 +346,58 @@ expected<ValueType, ErrorType>::and_then(const cxx::function_ref<void(ValueType&
     return *this;
 }
 
+template <typename ValueType, typename ErrorType>
+template <typename Optional = ValueType, typename std::enable_if<is_optional<Optional>::value, int>::type = 0>
```

Comment:
@elfenpiff Can the 2nd template parameter be omitted as it's already in the hpp? I remember we did in some first version of the `WaitSet`.